base/partr.jl

```diff
@@ -0,0 +1,164 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+module Partr
+
+# a task heap
+mutable struct taskheap
+    const lock::ReentrantLock
```

Comment:
We only use `trylock` so I guess it's fine, but isn't it a bit weird to use a lock that requires the scheduler for implementing the scheduler? Actually, why not just use a single atomic bool? I think it's fine as long as we have more heaps than the number of threads.

Reply:
Good point. I would actually like to make this an `InvasiveLinkedListSynchronized{Task}` (the same type as `StickyWorkqueue`). But for the initial port, `Threads.SpinLock` would be a more direct translation.